Multisig/Cross-layer/App.swift

```diff
@@ -53,5 +51,23 @@ class App {
 
     let firebaseConfig = FirebaseConfig()
 
-    private init() { }
+    private init() {
+        let updateNotifications: [NSNotification.Name] = [
+            .selectedSafeChanged, .selectedSafeUpdated, .networkInfoChanged
+        ]
+
+        for name in updateNotifications {
+            NotificationCenter.default.addObserver(self,
+                                           selector: #selector(bindNetworkInfo),
+                                           name: name,
+                                           object: nil)
+        }
+    }
+
+    @objc func bindNetworkInfo() {
+        if let network = try? Safe.getSelected()?.network {
+            clientGatewayService.bindNetwork(Int(network.chainId))
```

Comment:
This approach may break monitoring functionality. Right now we monitor pending WalletConnect transactions, but in future we may monitor more things - e.g. if we add tx execution functionality we might monitor tx status.
On the other hand we can add optional `network` parameter in future.